extmod/modusocket.c

```diff
@@ -155,6 +168,10 @@ STATIC mp_obj_t socket_accept(mp_obj_t self_in) {
     // new socket has valid state, so set the NIC to the same as parent
     socket2->nic = self->nic;
     socket2->nic_type = self->nic_type;
+    // set the same address family, socket type and protocol as parent.
+    socket2->domain = self->domain;
+    socket2->type = self->type;
+    socket2->proto = self->proto;
```

Comment:
I think this should be moved up, to before the `->accept()` call.  This is because `wiznet5k_socket_accept()` sets these values and does something tricky with `self->domain`, and I think this change will break that.

Reply:
Fixed.